code/ship/shiphit.cpp

```diff
@@ -2448,14 +2448,38 @@ static void ship_do_damage(object *ship_objp, object *other_obj, vec3d *hitpos,
 				apply_diff_scale = false;
 			}
 		}
+
+		// if weapon is vampiric, slap healing on shooter instead of target
+		weapon_info_index = shiphit_get_damage_weapon(other_obj);
+
+		if (weapon_info_index >= 0) {
+			weapon_info* wip = &Weapon_info[weapon_info_index];
+
+			if ((wip->wi_flags[Weapon::Info_Flags::Vampiric]) && (other_obj->parent > 0)) {
+				object* parent = &Objects[other_obj->parent];
+
+				if ((parent->type == OBJ_SHIP) && (parent->signature == other_obj->parent_sig)) {
+					ship* shipparent = &Ships[parent->instance];
```

Comment:
So two more style things you should be aware of.  There are lots of different cases that we use in FSO for different variable scopes, and some files even differ on that.  For locals, though, we use snake_case, and if you're in a file that uses it for locals, you should follow that convention.  Again some files might do other things.  When you're in those files, just follow what's done there.

The other style thing is that if a variable is a pointer, you should make it part of the variable name.  For instance, you'll see `objp` all over the place in FSO because it means object pointer.  `shipp` means ship pointer.  These pointers are pretty short lived, though, so it's not that big a deal.

I definitely think you should switch the case of shipparent, but I think that's the only thing I'm going to require.